internal/services/communication/email_service_domain_resource.go

```diff
@@ -0,0 +1,319 @@
+			}
+
+			if err := client.CreateOrUpdateThenPoll(ctx, id, param); err != nil {
+				return fmt.Errorf("creating %s: %+v", id, err)
+			}
+
+			metadata.SetID(id)
+
+			return nil
+		},
+	}
+}
+
+func (r EmailCommunicationServiceDomainResource) Update() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.Communication.DomainClient
+
+			var model EmailCommunicationServiceDomainResourceModel
+
+			if err := metadata.Decode(&model); err != nil {
+				return err
+			}
+
+			id, err := domains.ParseDomainID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			existing, err := client.Get(ctx, *id)
+			if err != nil || existing.Model == nil {
+				return fmt.Errorf("retrieving %s: %+v", *id, err)
+			}
+
+			domain := *existing.Model
+
+			props := pointer.From(domain.Properties)
+
+			if metadata.ResourceData.HasChange("user_engagement_tracking") {
+				userEngagementTracking := domains.UserEngagementTrackingDisabled
+				if model.UserEngagementTracking {
+					userEngagementTracking = domains.UserEngagementTrackingEnabled
+				}
+
+				props.UserEngagementTracking = pointer.To(userEngagementTracking)
+			}
+
+			if metadata.ResourceData.HasChange("tags") {
+				domain.Tags = pointer.To(model.Tags)
+			}
+
+			domain.Properties = &props
+
+			if err := client.CreateOrUpdateThenPoll(ctx, *id, domain); err != nil {
+				return fmt.Errorf("updating %s: %+v", id, err)
+			}
+
+			return nil
+		},
+	}
+}
+
+func (EmailCommunicationServiceDomainResource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.Communication.DomainClient
+
+			state := EmailCommunicationServiceDomainResourceModel{}
+
+			id, err := domains.ParseDomainID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			resp, err := client.Get(ctx, *id)
+			if err != nil {
+				if response.WasNotFound(resp.HttpResponse) {
+					return metadata.MarkAsGone(id)
+				}
+
+				return fmt.Errorf("retrieving %s: %+v", id, err)
+			}
+
+			state.Name = id.DomainName
+			state.ResourceGroupName = id.ResourceGroupName
+			state.EMailServiceName = id.EmailServiceName
+
+			if model := resp.Model; model != nil {
+				if props := model.Properties; props != nil {
+					state.DomainManagement = string(props.DomainManagement)
+
+					if props.FromSenderDomain != nil {
+						state.FromSenderDomain = pointer.From(props.FromSenderDomain)
+					}
+
+					if props.MailFromSenderDomain != nil {
+						state.MailFromSenderDomain = pointer.From(props.MailFromSenderDomain)
+					}
+
+					if props.UserEngagementTracking != nil {
+						state.UserEngagementTracking = false
+						if *props.UserEngagementTracking == domains.UserEngagementTrackingEnabled {
+							state.UserEngagementTracking = true
+						}
+					}
```

Comment:
this can be simplified to:
```suggestion					
					if props.UserEngagementTracking != nil {
						state.UserEngagementTracking = *props.UserEngagementTracking == domains.UserEngagementTrackingEnabled

					}
```